realm/realm-library/src/main/java/io/realm/RealmConfiguration.java

```diff
@@ -672,6 +692,30 @@ public Builder readOnly() {
             return this;
         }
 
+        /**
+         * Setting this will cause the Realm to compact the Realm file before returned to the user if the Realm file
```

Comment:
Don't mention specific details here as they might well change, and we will surely forget to update this. Perhaps something like:

```
Setting this will cause Realm to compact the Realm file if the Realm file has grown too large and a significant amount of space can be recovered. See {@link DefaultCompactOnLaunchCallback} for details.
``